pennylane/ops/op_math/adjoint_class.py

```diff
@@ -0,0 +1,305 @@
+from pennylane.math import transpose, conj
+
+
+# pylint: disable=no-member
+class AdjointOperation(Operation):
+    """This mixin class is dynamically added to an ``Adjoint`` instance if the provided base class is an ``Operation``.
+
+    .. warning::
+        This mixin class should never be initialized independent of ``Adjoint``.
+
+    Overriding the dunder method ``__new__`` in ``Adjoint`` allows us to customize the creation of an instance and dynamically
+    add in parent classes.
+
+    .. note:: Once the ``Operation`` class does not contain any unique logic any more, this mixin class can be removed.
+    """
+
+    # This inverse behavior only needs to temporarily patch behavior until in-place inversion is removed.
+
+    @property
+    def _inverse(self):
+        return self.base._inverse  # pylint: disable=protected-access
+
+    @_inverse.setter
+    def _inverse(self, boolean):
+        self.base._inverse = boolean  # pylint: disable=protected-access
+        # refresh name as base_name got updated.
+        self._name = f"Adjoint({self.base.name})"
+
+    def inv(self):
+        self.base.inv()
+        # refresh name as base_name got updated.
+        self._name = f"Adjoint({self.base.name})"
+        return self
+
+    @property
+    def base_name(self):
+        return self._name
+
+    @property
+    def name(self):
+        return self._name
+
+    # pylint: disable=missing-function-docstring
+    @property
+    def basis(self):
+        return self.base.basis
+
+    @property
+    def control_wires(self):
+        return self.base.control_wires
+
+    def single_qubit_rot_angles(self):
+        omega, theta, phi = self.base.single_qubit_rot_angles()
+        return [-phi, -theta, -omega]
+
+    @property
+    def grad_method(self):
+        return self.base.grad_method
+
+    # pylint: disable=missing-function-docstring
+    @property
+    def grad_recipe(self):
+        return self.base.grad_recipe
+
+    def get_parameter_shift(self, idx):
+        return self.base.get_parameter_shift(idx)
+
+    @property
+    def parameter_frequencies(self):
+        return self.base.parameter_frequencies
+
+    def generator(self):
+        return -1.0 * self.base.generator()
+
+
+# pylint: disable=too-many-public-methods
+class Adjoint(Operator):
+    """
+    The Adjoint of an operator.
+
+    Args:
+        base (~.operation.Operator): The operator that is adjointed.
+
+    .. seealso:: :func:`~.adjoint`, :meth:`~.operation.Operator.adjoint`
+
+    This is a *developer*-facing class, and the :func:`~.adjoint` transform should be used to construct instances
+    of this class.
+
+    **Example:**
+
+    >>> op = Adjoint(qml.S(0))
+    >>> op.name
+    'Adjoint(S)'
+    >>> qml.matrix(op)
+    array([[1.-0.j, 0.-0.j],
+       [0.-0.j, 0.-1.j]])
+    >>> qml.generator(Adjoint(qml.RX(1.0, wires=0)))
+    (PauliX(wires=[0]), 0.5)
+    >>> Adjoint(qml.RX(1.234, wires=0)).data
+    [1.234]
+
+    **Developer Details:**
```

Comment:
We now have the power to do
```rest
    .. details::
        :title: Developer Details
```

and it will create a dropdown box like Usage details. Perhaps worth starting to do?

Reply:
```suggestion
    .. details::
        :title: Developer Details
```